### PR TITLE
[PJRT] add platfrom_version for rocm in pjrt/gpu_device

### PR DESCRIPTION
**Pull request by reza-amd**
This PR fixes the returned value from platform_version() for ROCm.
@hawkinsp @deven-amd

### CODE REVIEW

File: tensorflow/compiler/xla/pjrt/gpu_device.cc [hunk 1]
```diff
@@ -166,13 +170,19 @@ class GpuClient : public xla::PjRtStreamExecutorClient {
       int num_replicas, int num_partitions) const override;
 
   absl::string_view platform_version() const override {
-#if defined(CUDART_VERSION)
+#if defined(TF_ROCM_VERSION) || defined(CUDART_VERSION)
```
Comment (deven-amd) [inline tensorflow/compiler/xla/pjrt/gpu_device.cc:173]: Can we make this conditional on `TENSORFLOW_USE_ROCM` only?

Comment (reza-amd) [reply tensorflow/compiler/xla/pjrt/gpu_device.cc]: Relying only TENSORFLOW_USE_ROCM is not a good idea, as we expect this from this function to attach a version number to "rocm "

**Review by hawkinsp — APPROVED** [review]

**Review by hawkinsp — APPROVED** [review]

**Comment by reza-amd** [comment]
@hawkinsp, could you please post the "feedback/copybara" failure log here.

**Comment by rthadur** [comment]
This is the error 

` error loading package 'third_party/tensorflow/compiler/xla/pjrt': Encountered error while reading extension file 'rocm/build_defs.bzl': no such package '@local_config_rocm//rocm': no such package '@local_config_rocm//': Accessing external repositories is not supported`

**Pull request merged**